crates/sui-bridge/src/crypto.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    error::{BridgeError, BridgeResult},
+    types::{BridgeCommittee, BridgeEvent, SignedBridgeEvent, VerifiedSignedBridgeEvent},
+};
+use fastcrypto::hash::Keccak256;
+use fastcrypto::secp256k1::{Secp256k1PublicKey, Secp256k1Signature};
+use serde::{Deserialize, Serialize};
+use std::{pin::Pin, sync::Arc};
+use sui_types::{crypto::Signer, message_envelope::VerifiedEnvelope};
+use tap::TapFallible;
+
+pub type BridgeAuthorityPublicKey = Secp256k1PublicKey;
+pub type BridgeAuthoritySignature = Secp256k1Signature;
+
+/// See `StableSyncAuthoritySigner`
+pub type StableSyncBridgeAuthoritySigner =
+    Pin<Arc<dyn Signer<BridgeAuthoritySignature> + Send + Sync>>;
+
+// TODO: include epoch ID here to reduce race conditions?
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct BridgeAuthoritySignInfo {
+    pub authority_pub_key: BridgeAuthorityPublicKey,
+    pub signature: BridgeAuthoritySignature,
+}
+
+impl BridgeAuthoritySignInfo {
+    pub fn new(
+        msg: &BridgeEvent,
+        authority_pub_key: BridgeAuthorityPublicKey,
+        secret: &fastcrypto::secp256k1::Secp256k1KeyPair,
+    ) -> Self {
+        let msg_bytes = msg.to_bytes();
+
+        Self {
+            authority_pub_key,
+            signature: secret.sign_with_hash::<Keccak256>(&msg_bytes),
+        }
+    }
+
+    pub fn verify(&self, msg: &BridgeEvent, committee: &BridgeCommittee) -> BridgeResult<()> {
```

Comment:
Could you remind me why we couldn't just use the Sui committee instead of a separate BridgeCommittee? I understand that at first we are not going to implement committee change for the bridge, but in practice we are only able to process messages from active sui validators anyway. So using Sui committee on the Sui side for any authentication should just work?

Reply:
The main reason is bridge key is secp256k1 but validator protocol key (part of Sui committee) is bls.